inc/wp-components/classes/integrations/class-google-tag-manager.php

```diff
@@ -94,31 +94,54 @@ public function get_post_id() : string {
 	}
 
 	/**
-	 * Get value for tags targeting.
+	 * Get array of taxnomy terms for data layer.
 	 *
+	 * @param  string $taxonomy Taxonomy for which terms should be extracted.
 	 * @return array
 	 */
-	public function get_tags() : array {
+	public function get_taxonomy_terms( $taxonomy ) : array {
 		// Single article.
 		if ( $this->query->is_single() ) {
-			$tags = wp_get_post_tags( $this->query->post->ID ?? 0 );
+			$terms = wp_get_post_terms( $this->query->post->ID, $taxonomy );
```

Comment:
If the taxonomy doesn't exist, `wp_get_post_terms()` returns a `\WP_Error`. It couldn't hurt to check for that and return an empty array early before doing the `array_map`

Reply:
Cood call, will do